flink-filesystems/flink-azure-fs-hadoop/pom.xml

```diff
@@ -119,17 +111,69 @@ under the License.
 								</includes>
 							</artifactSet>
 							<relocations>
+								<!-- relocate the references to Hadoop to match the shaded Hadoop config -->
 								<relocation>
 									<pattern>org.apache.hadoop</pattern>
-									<shadedPattern>org.apache.flink.fs.shaded.hadoop.org.apache.hadoop</shadedPattern>
+									<shadedPattern>org.apache.flink.fs.shaded.hadoop3.org.apache.hadoop</shadedPattern>
 								</relocation>
-								<!-- relocate the azure-storage dependencies -->
+
+								<!-- shade dependencies internally used by Hadoop and never exposed downstream -->
+								<relocation>
+									<pattern>org.apache.commons</pattern>
+									<shadedPattern>org.apache.flink.fs.shaded.hadoop3.org.apache.commons</shadedPattern>
+								</relocation>
+
+								<!-- relocate the Azure dependencies -->
 								<relocation>
-									<pattern>com.microsoft.azure.storage</pattern>
-									<shadedPattern>org.apache.flink.fs.shaded.com.microsoft.azure.storage</shadedPattern>
+									<pattern>com.microsoft.azure</pattern>
+									<shadedPattern>org.apache.flink.fs.azure.shaded.com.microsoft.azure</shadedPattern>
+								</relocation>
+
+								<!-- shade dependencies internally used by Azure and never exposed downstream -->
+								<relocation>
+									<pattern>org.apache.httpcomponents</pattern>
```

Comment:
do we still need all this now that we have plugins?

Reply:
If the plugins are used then not. If people drop the jar into `lib` it would still be beneficial. Therefore, I opted for this approach.